sys/vm_map.c

```diff
@@ -166,16 +166,23 @@ int vm_map_findspace(vm_map_t *map, vm_addr_t start, size_t length,
     vm_map_entry_t *next = TAILQ_NEXT(it, map_list);
     if (!next)
       continue;
-    if (next->start - it->end >= length) {
-      /* We will fit into this gap. */
+    if (start < it->end && next->start - it->end >= length) {
```

Comment:
Can't you simplify both cases to a single expression: `(start >= it->end) && (start + length <= next->start)`?